performance_regression_cdc_test.py

```diff
@@ -0,0 +1,294 @@
+
+        node1: BaseNode = self.db_cluster.nodes[0]
+
+        with self.cql_connection_patient(node1) as session:
+            session.execute(f"TRUNCATE TABLE {self.keyspace}.{self.table}")
+            session.execute(f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_enabled")
+        self.check_regression_with_baseline(base_line="cdc_disabled")
+
+    def test_write_with_cdc_pre_image(self):
+        write_cmd = self.params.get("stress_cmd_w")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_disabled")
+        node1: BaseNode = self.db_cluster.nodes[0]
+
+        with self.cql_connection_patient(node1) as session:
+            session.execute(f"TRUNCATE TABLE {self.keyspace}.{self.table}")
+            session.execute(f"TRUNCATE TABLE {self.keyspace}.{self.table}_scylla_cdc_log")
+            session.execute(
+                f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true, 'preimage': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_preimage_enabled")
+
+        self.check_regression_with_baseline(base_line="cdc_disabled")
+
+    def test_write_profile(self):
+        self.keyspace = "cdc_keyspace"
+        self.table = "test_cdc"
+        write_cmd = self.params.get("stress_cmd_w")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_disabled")
+
+        node1: BaseNode = self.db_cluster.nodes[0]
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}", timeout=300)
+        except (UnexpectedExit, Failure) as details:
+            self.log.warning("Trancate error %s. Sleep and continue", details)
+            time.sleep(60)
+
+        node1.run_cqlsh(f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_enabled")
+
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+
+        try:
+            node1.run_cqlsh(f"TRUNCATE TABLE {self.keyspace}.{self.table}", timeout=300)
+        except (UnexpectedExit, Failure) as details:
+            self.log.warning("Trancate error %s. Sleep and continue", details)
```

Comment:
Currently there is a bug of truncate : https://github.com/scylladb/scylla/issues/4924

If the old data isn't cleaned for timeout exception, will it affect the performance result?
If it affected, we need to retry truncate or clean the old data by other method.

Reply:
This is bad news. Need to find new way. Don't think it would affect on performance.